variod.c

```diff
@@ -283,8 +285,7 @@ int main(int argc, char *argv[])
 
```

Comment:
My system only gets here once after start up. I see "Connection accepted" only once at the beginning of the log file.
That's why I think something else must be going on.

Reply:
Here's the problem:
You mute at the bottom of the while and unmute here, constantly.
If you infinitely enable audio you can never mute it.

```
	while(1) {
		//Accept and incoming connection
		fprintf(fp_console,"Waiting for incoming connections...\n");
		fflush(fp_console);
		c = sizeof(struct sockaddr_in);

		//accept connection from an incoming client
		connfd = accept(listenfd, (struct sockaddr *)&s_xcsoar, (socklen_t*)&c);
		if (connfd < 0) {
			fprintf(stderr, "accept failed");
			return 1;
		}

		fprintf(fp_console, "Connection accepted\n");

		// Socket is connected

		//enable vario sound
		vario_unmute();
		// get current values for Polar, MC, ... from XCSoar
		// since we might have started after XCSoar was running for a while
		request_current_settings = true;

		//Receive a message from sensord and forward to XCsoar
		while ((read_size = recv(connfd, client_message, 2000, 0 )) > 0 ) {
```
I'm using real sensors.
Sensord has a specific timing. It doesn't always send a message.
If your bot sends messages constantly it never gets out of the receiving loop.
Try missing some messages in your bot and the problem appears.

Reply:
Don't know what you mean "missing some messages".
My bot sends 1 NMEA sentence every second. I can slow it down as much as I want: no difference. recv() doesn't return until it has something no matter how long it has to wait. But if recv() detects a broken connection while it waits it returns -1. That is probably what's happening. Moving unmute up cures the symptom, but not the root cause. And the root cause is probably in sensord. And I don't think I want to open the lid on that.

Reply:
That makes sense.
In the actual device Variod.log does not show as clean as your logs.
I has reconnections all over.
We'll have to dive into sensord if this is going to be solved...